### ISSUE TITLE
'NoneType' object has no attribute 'handle_message'

### ISSUE THREAD

**Issue by PrasannaGaneshu** [RasaHQ/rasa_core#1709]
**Rasa Core version**: 0.12.3
   RASA NLU :: 0.13.8

**Python version**: 3.6.1

**Operating system** (windows, osx, ...):

**Issue**:  'NoneType' object has no attribute 'handle_message'

Below is code snippet for server.py file which is causing an error ::
```python
`@app.route('/api/v1/<sender_id>/respond', methods=['GET', 'POST'])

    def respond(self, request, sender_id):
        request.setHeader('Content-Type', 'application/json')
        request.setHeader('Access-Control-Allow-Origin', '*')
        
        request_params = request_parameters(request)
        print("=========Request PARam================: ",request_params)
        if 'query' in request_params:
        	message = request_params.pop('query')
        elif 'q' in request_params:
        	message = request_params.pop('q')
        else:
        	request.setResponseCode(400)
        	return json.dumps({"error": "Invalid parse parameter specified"})
        try:
            out = CollectingOutputChannel()
            print("***********", message, "====", out, "====", sender_id)
            response = self.agent.handle_message(message, output_channel=out, sender_id=sender_id) #response = UserMessage(message, out, sender_id)
            #processor = self.agent._create_processor()
            #response = processor.handle_message(UserMessage(message, out, sender_id))
            print("==UserMessage==", response.__dict__)
            request.setResponseCode(200)
            print("TRY: " , response)
            return json.dumps(response)
        except Exception as e:
            request.setResponseCode(500)
            logger.error("Caught an exception during parse: {}".format(e), exc_info=1)
            return json.dumps({"error": "{}".format(e)})
		
```
Below is the error screenshot 
![error_feb11 2](https://user-images.githubusercontent.com/38690247/52558376-354f6800-2e18-11e9-8b88-8e3b3fd88608.png)

**Comment by akelad** [comment]
Thanks for raising this issue, @wochinge  will get back to you about it soon.

**Comment by wochinge** [comment]
@PrasannaGaneshu What are you trying to do? Could you please give me a bit of context?

**Comment by PrasannaGaneshu** [comment]
Hi,I created a weather it similar to @Juste weatherbot tutorial using version 7 of RASA. However I happened to upgrade RASA to version 12,after which I'm getting empty response from this agent.

The code snippet in my previous reply is from server.py file

**Comment by wochinge** [comment]
@PrasannaGaneshu Where are you creating the `agent` object in your code?
Please don't tag other people in the issue without reason. We will coordinate the issue handling and make sure that we help you to find out whether this a bug.

**Comment by PrasannaGaneshu** [comment]
Sorry for that!I have posted multiple times regarding this 'agent' issue but i could not get a proper response so happened to tag other people from RASA.

I've created the `agent` object in server.py file...Im attaching the file for reference..

`import logging
from rasa_core.agent import Agent
from rasa_core.interpreter import RasaNLUInterpreter
from rasa_core.channels.channel import CollectingOutputChannel
from klein import Klein
import datetime
import urllib3
http=urllib3.PoolManager()
import json
import requests
import time
import calendar
from ast import literal_eval
import os

logger = logging.getLogger(__name__)

def request_parameters(request):
	if request.method.decode('utf-8', 'strict') == 'GET':
		return {
			key.decode('utf-8', 'strict'): value[0].decode('utf-8','strict')
			for key, value in request.args.items()}
	else:
		content = request.content.read()
		try:
			return json.loads(content.decode('utf-8', 'strict'))
		except ValueError as e:
			logger.error("Failed to decode json during respond request. Error: {}. Request content: '{}' ".format(e, content))
			raise


class Server:
	app = Klein()

	def __init__(self, model_directory, interpreter):
		self.model_directory = model_directory
		self.interpreter = interpreter
		self.agent = self._create_agent(model_directory, interpreter)

	@staticmethod
	def _create_agent(model_directory, interpreter):
		"""Creates a Rasa Agent which runs when the server is started"""
		try:
			return Agent.load(model_directory, interpreter)
		except Exception as e:
			logger.warn("Failed to load any agent model. Running Rasa Core server with out loaded model now. {}".format(e))
			return None
            
	@app.route("/api/v1/status", methods=['GET'])
	def status(self, request):
		"""Check if the server is running and responds with the status."""
		request.setHeader('Access-Control-Allow-Origin', '*')
		return json.dumps({'status': 'OK'})
        
        

	@app.route('/api/v1/<sender_id>/parse', methods=['GET', 'POST'])
	def parse(self, request, sender_id):
		request.setHeader('Content-Type', 'application/json')
		request_params = request_parameters(request)
		if 'query' in request_params:
			message = request_params.pop('query')
		elif 'q' in request_params:
			message = request_params.pop('q')
		else:
			request.setResponseCode(400)
			return json.dumps({"error": "Invalid parse parameter specified"})
		try:
			response = self.agent.start_message_handling(message, sender_id)
			request.setResponseCode(200)
			return json.dumps(response)
		except Exception as e:
			request.setResponseCode(500)
			logger.error("Caught an exception during parse: {}".format(e), exc_info=1)
			return json.dumps({"error": "{}".format(e)})
			

	@app.route('/api/v1/<sender_id>/respond', methods=['GET', 'POST'])
	def respond(self, request, sender_id):
		request.setHeader('Content-Type', 'application/json')
		request.setHeader('Access-Control-Allow-Origin', '*')
        request_params = request_parameters(request)
		if 'query' in request_params:
			message = request_params.pop('query')
		elif 'q' in request_params:
			message = request_params.pop('q')
		else:
			request.setResponseCode(400)
			return json.dumps({"error": "Invalid parse parameter specified"})
		try:
			out = CollectingOutputChannel()
			response = self.agent.handle_message(message, output_channel=out, sender_id=sender_id)
			request.setResponseCode(200)
			return json.dumps(response)
		except Exception as e:
			request.setResponseCode(500)
			logger.error("Caught an exception during parse: {}".format(e), exc_info=1)
			return json.dumps({"error": "{}".format(e)})


			
		
if __name__ == "__main__":
    server = Server("models/dialogue", RasaNLUInterpreter("models/nlu/default/weathernlu"))
    server.app.run("0.0.0.0", 8087)
`

**Comment by wochinge** [comment]
1. Is there a exception and what does it say for this part

```python
except Exception as e:
			logger.warn("Failed to load any agent model. Running Rasa Core server with out loaded model now. {}".format(e))
			return None
```

2. May I ask what you are trying to do? So why are you writing the custom code if you can also use the command line interface?

3. You can format code blocks with ``` (in the line before and after the code block). A single ` is more for inline formatting

**Comment by PrasannaGaneshu** [comment]
1.We are not getting any exception.
2 .I've created a chat bot using version 7 of RASA, It worked perfectly fine.I tried to upgrade RASA to Version 13 (rasa-nlu version: 0.14.1 ,rasa-core version: 0.13.0)
I wanted to migrate it to version 13 with less changes possible. Ive created a frontend UI for which Rest API's needs to be exposed.The custom code is already written while I was working with Version 7 of RASA,Now I tried to upgrade RASA,it shows an error "None Type Object has no attribute "handle_message"

**Comment by wochinge** [comment]
@PrasannaGaneshu 
1) I recommend to use the [REST channel] (https://rasa.com/docs/core/connectors/#rest-channels) of Rasa Core in your UI instead of rewriting the endpoints.

2) Did you retrain NLU? Seems you need more training data

3) Since this is not a bug within Rasa Core / NLU, I close this issue now. Please feel welcome to get help in the [community forum](https://forum.rasa.com).

**Comment by PrasannaGaneshu** [comment]
@wochinge ,
My issue is not solved....
1.I want to migrate from version 7 to version 13 of RASA which less changes possible.So kept the channel as it is.
2.I happened to re-train NLU.I have close to 50 examples for the 3 intents i used in my bot.
3.I happened to resolve a minor issue(User_id not getting captured in the slot) caused due to lack of proper training.

My endpoint server is up and running .The bot works just fine from the command line. I..But im getting empty response when im trying to hit it through postman..
Can someone help me out with this..
Here is my error_screenshot in postman:

![error_postman](https://user-images.githubusercontent.com/38690247/52949995-e0759980-33a3-11e9-93b3-25a1a38baecb.png)
